New FastCGI module for PHP with Apache (#613)
CleverCloud/doc.clever-cloud.com: partials/language-specific-deploy/php.md

DirectorySlash Off
RewriteEngine On
RewriteCond %{REQUEST_FILENAME} -d
RewriteRule ^(.+[^/])$ %{HTTP:X-Forwarded-Proto}://%{HTTP_HOST}/$1/ [R=301,L,QSA]
```

These statements will keep the former protocol of the request when issuying the redirect. Assuming that the header X-Forwarded-Proto is always filled (which is the case on our platform).

If you want to force all redirects to HTTPS, you can replace `%{HTTP:X-Forwarded-Proto}` with `https`.

> ### Change FastCGI module

Reviewer: ```suggestion
### Change the FastCGI module
```

You can choose between two FastCGI modules, `fastcgi` and `proxy_fcgi`.

> To choose between this two modules you must use `CC_CGI_IMPLEMENTATION` environment variable with `fastcgi` or `proxy_fcgi` as value.

Reviewer: ```suggestion
To choose between this two modules you must use the `CC_CGI_IMPLEMENTATION` environment variable with `fastcgi` or `proxy_fcgi` as a value.
```

Reviewer 2: s/this/these/

{{ < alert "warning" > }}

> We recommend to use `proxy_fcgi` over the `fastcgi` implementation because it is not maintained anymore but to prevent unexpected behaviors with historical applications it has been kept as the default one.

Reviewer: ```suggestion
We recommend preferring `proxy_fcgi` over `fastcgi`. The `fastcgi` implementation is not maintained anymore, but has been kept as default to prevent unexpected behaviors with historical applications.
```

{{ < /alert > }}

> If you have issues with content downloading, it could be related to `fastcgi` module which is not working correctly in combination with `deflate` module as the `Content-Length` header is not updated to the new size of the encoded content.

Reviewer: ```suggestion
If you have issues with downloading content, it could be related to the `fastcgi` module not working correctly in combination with the `deflate` module, as the `Content-Length` header is not updated to the new size of the encoded content.
```

> To resolve this issue, we advise you to change default FastCGI module to `proxy_fcgi`.

Reviewer: ```suggestion
To resolve this issue, we advise you to switch the value of `CC_CGI_IMPLEMENTATION` from default to `proxy_fcgi`.
```

## Using HTTP authentication

> Using basic HTTP authentication, PHP usually gives you the user and password in variables named `$_SERVER['PHP_AUTH_USER']` and `$_SERVER['PHP_AUTH_PW']`.

Reviewer: ```suggestion
Using basic HTTP authentication, PHP usually handles the values of user and password in variables named `$_SERVER['PHP_AUTH_USER']` and `$_SERVER['PHP_AUTH_PW']`.
```
the "you" is useless in this context
"user" and "password" are better defined as values, for clarity

> On Clever Cloud, we have enabled an option in Apache to pass directly the Authorization header even though we are using FastCGI; still, PHP does not use the header and the variables mentioned before are empty.

Reviewer: ```suggestion
At Clever Cloud, we have enabled an Apache option to pass directly the Authorization header, even though we are using FastCGI; still, the header is not used by PHP, and the aforementioned variables are empty.
```

Because this variant: 

> PHP does not use the header and the variables…

implies for a split  second of the reader's attention that both header and variables are objects sharing the same fate of not being used by PHP. 
Using the passive voice solves the issue.
